### PR TITLE
Added new commands for quitting, opening tabs and added arguments

### PR DESCRIPTION
**Pull request by fijter**
Great to have the base ready, I extended it with some other much used commands including wq, q, quit, tabe, tabedit and I added the option to apply arguments to ex-commands so you can open files if you know the full path without a dialog.

### CODE REVIEW

File: lib/ex.coffee [hunk 1]
```diff
@@ -1,9 +1,30 @@
...
+  q: =>
+    @quit()
+  tabedit: (filenames) ->
+    if filenames? and filenames.length > 0
+      atom.open(pathsToOpen: filenames)
+    else
+      atom.open()
+  tabe: (filenames) =>
+    @tabedit(filenames)
+  write: (close=false) =>
```
Comment (AsaAyers) [inline lib/ex.coffee]: Seems like mixing user supplied parameters (`:tabe <filename>`) and internal parameters like this `close` might lead to problems. Maybe instead any potentially asyncronous operations here should return promises.

One possible implementation might look like this.

```CoffeeScript
    write: =>
        dfd = Promise.defer()
        if atom.workspace.getActiveTextEditor().getPath() isnt undefined
            # not sure of the exact implementation. Just trying to put this together in a github comment.
        else
            atom.workspace.getActivePane().saveActiveItemAs(dfd.resolve)
        dfd.promise
        

    wq: =>
        @write().then(=> @quit())
````

Comment (fijter) [inline lib/ex.coffee]: You are absolutely right, thanks for your feedback :)

**Comment by lloeki** [comment]
@fijter that's exactly what I thought of. Thanks for the implementation! The `split` part should ultimately become a richer parser, so as to handle ranges (incl. markers).

Great point @AsaAyers. I like your example. I was thinking about using the return value for stuff like `%!` but had some bad feeling about it, that's why I didn't go further and wanted to get some feedback from people more apt at JS than me.

**Comment by lloeki** [comment]
Merged by 55bb9f631907b3d49e90f830adbe9d33a6049348. I fixed a number of things in 66e5f697f6cfe7f9b5ef4a27ed8939e56d07cbb3 to make it more conformant with vim.

I removed `:wq` to make argument passing consistent, keeping promises for a future changeset. Also, arguments are now passed with a splat.

**Comment by lloeki** [comment]
`:wq` with promises has been added in 4adc1c168a277f03edcf1a3c5dac2a82e46d8391

**Pull request closed without merging**